src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,138 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.controller.LinearQuadraticRegulator;
+import edu.wpi.first.math.estimator.KalmanFilter;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.system.LinearSystem;
+import edu.wpi.first.math.system.LinearSystemLoop;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.Utils;
+
+import static frc.robot.Constants.*;
+import static frc.robot.Constants.Shooter.*;
+import static frc.robot.Ports.Shooter.*;
+
+public class Shooter extends SubsystemBase {
+    private static Shooter INSTANCE;
+    private final UnitModel unitModel = new UnitModel(TICKS_PER_RADIAN);
+    private final WPI_TalonFX mainMotor = new WPI_TalonFX(MAIN_MOTOR);
+    private final LinearSystemLoop<N1, N1, N1> linearSystemLoop;
+    private double currentTime = 0;
+    private double lastTime = 0;
+
+    /**
+     * Constructor.
+     */
+    private Shooter() {
+        CONFIGURATION.enableOptimizations = true;
+        mainMotor.configAllSettings(CONFIGURATION);
+        mainMotor.setInverted(IS_MAIN_INVERTED);
+        mainMotor.setSensorPhase(MAIN_SENSOR_PHASE);
+        mainMotor.configNeutralDeadband(NEUTRAL_DEADBAND, Constants.TALON_TIMEOUT);
+        linearSystemLoop = configStateSpace(true);
+    }
+
+    /**
+     * Gets the single instance of the shooter.
+     *
+     * @return shooter instance.
+     */
+    public static Shooter getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Shooter();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * State space configuration function. Note that there are 2 different configurations.
+     *
+     * @param isIneritaBased is the configuration for the state space.
+     *                       If the value is true, the state space is based off of an inertia model.
+     *                       Otherwise, the state space is based off of a voltage equation model.
+     * @return the linear system loop (based on the type).
+     */
+    private LinearSystemLoop<N1, N1, N1> configStateSpace(boolean isIneritaBased) {
+        final DCMotor motor = DCMotor.getFalcon500(1);
+
+        LinearSystem<N1, N1, N1> flywheel_plant;
+        if (!isIneritaBased)
+            flywheel_plant = new LinearSystem<>(A_KaKv, B_KaKv, C_KaKv, D_KaKv);
```

Comment:
You can use [this](https://github.com/wpilibsuite/allwpilib/blob/8c55844f9177a02bf0f2c93344c2e7309f7804c0/wpimath/src/main/java/edu/wpi/first/math/system/plant/LinearSystemId.java#L192) function instead